scvi/models/autozivae.py

```diff
@@ -0,0 +1,312 @@
+import torch
+import torch.nn.functional as F
+from torch.distributions import Normal, Beta, Gamma, kl_divergence as kl
+
+
+from scvi.models.log_likelihood import log_zinb_positive, log_nb_positive
+from scvi.models.vae import VAE
+from scipy.special import logit
+from scvi.models.utils import one_hot
+
+torch.backends.cudnn.benchmark = True
+
+
+class AutoZIVAE(VAE):
+    def __init__(
+        self,
+        n_input: int,
+        alpha_prior: float = 0.5,
+        beta_prior: float = 0.5,
+        minimal_dropout: float = 0.01,
+        zero_inflation="gene",
+        **args,
+    ):
+
+        if "reconstruction_loss" in args:
+            raise ValueError(
+                "No reconstruction loss must be specified for AutoZI : it is 'autozinb'."
+            )
+
+        super().__init__(n_input, **args)
+        self.zero_inflation = zero_inflation
+        self.reconstruction_loss = "autozinb"
+        self.minimal_dropout = minimal_dropout
+
+        # Parameters of prior Bernoulli Beta distribution : alpha + beta = 1 if only one is specified
+        if beta_prior is None and alpha_prior is not None:
+            beta_prior = 1.0 - alpha_prior
+        if alpha_prior is None and beta_prior is not None:
+            alpha_prior = 1.0 - beta_prior
+
+        device = (
+            torch.device("cuda") if torch.cuda.is_available() else torch.device("cpu")
```

Comment:
this should not be necessary these lines. See line 86 in trainer.py

Reply:
In fact, model.cuda() was not putting the prior alpha and beta on CUDA when they are fixed and when applying model.cuda() in trainer.py. But if we ever choose `use_cuda=False` when a GPU is available, we would obtain prior alpha and beta on CUDA and the rest on CPU. The best is to make a new `cuda` method tackling the discrepancy with prior alpha and beta directly.